openedx/features/smart_referral/api_views.py

```diff
@@ -0,0 +1,41 @@
+from django.http import JsonResponse
+from edx_rest_framework_extensions.auth.session.authentication import SessionAuthenticationAllowInactiveUser
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.views import APIView
+
+from .helpers import (
+    get_email_domain,
+    get_org_admin_email,
+    sort_contacts_by_org_and_user_domain,
+    get_platform_contacts_and_non_platform_contacts
+)
+
+
+class FilterContactsAPIView(APIView):
+    """
+    FilterContactsAPIView is used to filter user's contacts based on some specific criteria
+    """
+    authentication_classes = (SessionAuthenticationAllowInactiveUser,)
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request):
+        user = request.user
+        user_contacts = request.data.get('filtered_contacts_list', '')
```

Comment:
This view will be receiving list of contacts so instead of `filtered_contacts_list`, here simple 'contacts_list' would be more appropriate.

Reply:
done